### ISSUE TITLE
Wrong router-link-active-classes on prerendered pages

### ISSUE THREAD

**Issue by codeflorist** [nuxt/framework#6226]
### Environment

- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-rc.6`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `css`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/nuxt-starter-ejjmoq?file=pages/page.vue

Then request the path `/page` directly in the browser-preview

### Describe the bug

When using `nuxt generate` the prerendered pages do not have the correct `router-link-active` and `router-link-exact-active` classes applied. 

It seems, that the classes for all pages are generated on the assumption that the current route is `/`, meaning `<NuxtLink to="/">` always has the active-classes, while links to the actual current route have not.

This is only when directly requesting the prerendered page (e.g. `/page`). Once one uses a link to navigate within the app, the classes are appllied correctly.

**Comment by danielroe** [comment]
The issue here is that `serve` is not returning `/page/index.html` when you are visiting `/page` but `/index.html`.

We likely need to remove spa fallback support for `nuxi preview` (or resolve issue upstream in https://github.com/vercel/serve).

**Comment by codeflorist** [comment]
> The issue here is that `serve` is not returning `/page/index.html` when you are visiting `/page` but `/index.html`.

Ah, i didn't realize that, sorry. Thanks for the clarification! So this should not be a problem in a live environment.

**Comment by danielroe** [comment]
Yes, this seems to be only present when previewing. For now, you can work around manually with `npx serve .output/public` (ie. without the `-s` option).
